bitcoin_transfer/spend_your_coin.md

```diff
@@ -83,60 +83,65 @@ Do you remember the main questions? **From where, to where and how much?**
 Constructing the **TxIn** and adding it to the transaction is the answer to the "from where" question.  
 Constructing the **TxOut** and adding it to the transaction is the answer to the remaining ones.
 
-The donation address of this book is: [1KF8kUVHK42XzgcmJF4Lxz4wcL5WDL97PB](https://blockchain.info/address/1KF8kUVHK42XzgcmJF4Lxz4wcL5WDL97PB)  
-This money goes into my "Coffee and Sushi Wallet" that will keep me fed and compliant while writing the rest of the book.  
-If you succeed in completing this challenge you will be able to find your contribution amongst the **Hall of the Makers** on [http://n.bitcoin.ninja/](http://n.bitcoin.ninja/) \(ordered by generosity\).
+> The donation address of this book is: [1KF8kUVHK42XzgcmJF4Lxz4wcL5WDL97PB](https://blockchain.info/address/1KF8kUVHK42XzgcmJF4Lxz4wcL5WDL97PB)  
+This money goes into Nicolas' "Coffee and Sushi Wallet" that will keep him fed and compliant while writing the rest of the book.  
+If you succeed in completing this challenge on the MainNet you will be able to find your contribution among the **Hall of the Makers** on [http://n.bitcoin.ninja/](http://n.bitcoin.ninja/) \(ordered by generosity\).
 
+To get the our MainNet address:
 ```cs
 var hallOfTheMakersAddress = new BitcoinPubKeyAddress("1KF8kUVHK42XzgcmJF4Lxz4wcL5WDL97PB");
 ```
 
-If you are working on the testnet, send the testnet coins to any testnet address.
+Or if you are working on the TestNet, send the TestNet coins to [our TestNet address](https://live.blockcypher.com/btc-testnet/address/mzp4No5cmCXjZUpf112B1XWsvWBfws5bbB/).
 
 ```cs
-var hallOfTheMakersAddress = BitcoinAddress.Create("mzp4No5cmCXjZUpf112B1XWsvWBfws5bbB");
+var hallOfTheMakersAddress = BitcoinAddress.Create("mzp4No5cmCXjZUpf112B1XWsvWBfws5bbB", Network.TestNet);
 ```
 
 ### How much?
 
-If you want to send **0.5 BTC** from a **transaction input** with **1 BTC** you actually have to spend it all!  
-As the diagram shows below, your **transaction output** specifies  **0.5** BTC to Hall of The Makers and **0.4999** back to you.  
-What happens to the remaining **0.0001 BTC**? This is the miner fee in order to incentivize them to add this transaction into their next block.
+Bitcoin has [several units to use](https://en.bitcoin.it/wiki/Units). 1 bitcoin (BTC) is 1'000 milli-bitcoins/millibits. 1 milli-bitcoin (mBTC) is 1'000 micro-bitcoins/bits. And finally 1 (&micro;BTC) micro-bitcoin is 100 satoshis. 1 satoshi (sat) is the smallest unit on the Bitcoin network.  
```

Comment:
The only ones those enjoys consensus is BTC and Satoshi.

Reply:
Yes, but all of those mentioned are also an option in Bitcoin Core and in NBitcoin as well, so I thought it is useful to introduce them here. @NicolasDorier , what do you think? Are you planning to keep these units in NBitcoin?

Reply:
Yeah mBTC is strange. What you should use and introduce is `Bits`. Which is gaining some consensus.

Reply:
I am utterly confused by mBTC, millibit, bit, microbtc, kilobtc proposals. They didn't get traction during the past 3 years, they will not get traction now, that people cannot even afford to use BTC frequently enough, let's not beat the dead horse. If you are keen to go with some smaller unit maybe consider @psztorc 's `finney` proposal.

Reply:
I disagree, I'm definitely confused by millibits, millibitcoins, microbitcoins et-al, but not by bits. Bits is easy, and I've heard some wallets use it by default already (bread?), so I agree with @NicolasDorier here.

Reply:
So, does this sound good to you: I change the unit introduction section to describe only the relationship between bitcoins, bits and satoshis (without confusing the milli/micro stuff) and I use bits in the example?

Reply:
It doesn't. Not even 1% of the ecosystem supports bits. I often read this book, please don't confuse me with occasional bits.